### PR TITLE
PLATE SPRITES

### PR DESCRIPTION
**Pull request by binarysudoku**
## About the PR 

New plate sprites
Plate stacks now go up to 8 or 13 on ASS_JAM
Fixed a bug where inhands dont update properly when you click a plate with a platestack

## Changelog

```
(u)Erinexx:
(+)Added new plate sprites and fixed a display bug.
```

### CODE REVIEW

File: code/obj/item/kitchen.dm [hunk 1]
```diff
@@ -981,20 +980,29 @@ TRAYS
 /obj/item/platestack
 	name = "Stack of Plates"
 	desc = "It's a stack of plates"
-	icon = 'icons/obj/foodNdrink/platestack.dmi' //temporary in case of the dmi being different on the live version :)
+	icon = 'icons/obj/foodNdrink/platestack.dmi' // temporary in case of the dmi being different on the live version :)
```
Comment (ZeWaka) [inline code/obj/item/kitchen.dm:983]: remove comment

File: code/obj/item/kitchen.dm [hunk 2]
```diff
@@ -520,13 +520,12 @@ TRAYS
...
 			var/obj/item/platestack/stack = W
-			if(stack.platenum >= 7)
+			if(stack.platenum >= stack.platemax)
 				boutput(user,"<span style=\"color:red\"><b>The plates are piled too high!</b></span>")
 				return
 			src.set_loc(user)
 			stack.platenum++
-			stack.icon_state = "platestack[stack.platenum]"
-			stack.item_state = "platestack[stack.platenum]"
+			stack.update_icon(user)  // moved this stuff into a callable proc because it wasn't updating the item state in here
```
Comment (ZeWaka) [inline code/obj/item/kitchen.dm:528]: comment seems unnecessary since this is standard practice

File: code/obj/item/kitchen.dm [hunk 3]
```diff
@@ -981,20 +980,29 @@ TRAYS
...
+		src.icon_state = "platestack[src.platenum]"
+		src.item_state = "platestack[src.platenum]"
+		user.update_inhands()
 
 	attackby(obj/item/weapon as obj,mob/user as mob)
 		if(istype(weapon,/obj/item/plate) && !(istype(weapon,/obj/item/plate/tray)))
 			var/obj/item/plate/p = weapon
 			if(!p.ordered_contents.len)
-				if(!(platenum >= 7))
+				if(!(platenum >= platemax))
```
Comment (ZeWaka) [inline code/obj/item/kitchen.dm:1003]: this looks to delete the plates every single attackby call - i think you meant 1 there?

Comment (binarysudoku) [reply code/obj/item/kitchen.dm:1003]: The change here is only replacing the original hardcoded limit of 7 with a variable - it evaluates the same as it did already and it still works

File: code/obj/item/kitchen.dm [hunk 4]
```diff
@@ -1005,30 +1013,26 @@ TRAYS
...
+			else if(src.platenum == platemax)
 				boutput(user,"<span style=\"color:red\"><b>The plates are piled too high!</b></span>")
 				return
 			else
 				src.platenum += (p.platenum+1)
-				src.icon_state = "platestack[src.platenum]"
-				src.item_state = "platestack[src.platenum]"
-			if(keeptrigger != 1)
+				src.update_icon(user)
+			if(keeptrigger != platemax)
```
Comment (ZeWaka) [inline code/obj/item/kitchen.dm:1028]: sorry, this is where I meant to comment - this will have some FUNK behaviour I think

Comment (binarysudoku) [reply code/obj/item/kitchen.dm:1028]: oh fuck
im not sure how i did that, heck

Comment (binarysudoku) [reply code/obj/item/kitchen.dm]: fixed in latest push

**Pull request merged**